src/library/serato/seratofeature.cpp

```diff
@@ -0,0 +1,1043 @@
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
+    return qFromBigEndian<quint32>(data);
+#else
+    return qFromBigEndian<quint32>(
+            reinterpret_cast<const uchar*>(data.constData()));
+#endif
+}
+
+inline bool parseTrack(serato_track_t& track, QIODevice& buffer) {
+    QByteArray headerData = buffer.read(8);
+    while (headerData.length() == 8) {
+        QString fieldName = QString(headerData.mid(0, 4));
+        quint32 fieldId = parseUInt32(headerData.mid(0, 4));
+        quint32 fieldSize = parseUInt32(headerData.mid(4, 8));
+
+        // Read field data
+        QByteArray data = buffer.read(fieldSize);
+        if (static_cast<quint32>(data.length()) != fieldSize) {
+            qWarning() << "Failed to read "
+                       << fieldSize
+                       << " bytes for "
+                       << fieldName
+                       << " field.";
+            return false;
+        }
+
+        // Parse field data
+        switch (static_cast<FieldId>(fieldId)) {
+        case FieldId::FileType:
+            track.filetype = parseText(data, fieldSize);
+            break;
+        case FieldId::FilePath:
+            track.location = parseText(data, fieldSize);
+            break;
+        case FieldId::SongTitle:
+            track.title = parseText(data, fieldSize);
+            break;
+        case FieldId::Artist:
+            track.artist = parseText(data, fieldSize);
+            break;
+        case FieldId::Album:
+            track.album = parseText(data, fieldSize);
+            break;
+        case FieldId::Genre:
+            track.genre = parseText(data, fieldSize);
+            break;
+        case FieldId::Length: {
+            bool ok;
+            int duration = parseText(data, fieldSize).toInt(&ok);
+            if (ok) {
+                track.duration = duration;
+            }
+            break;
+        }
+        case FieldId::Bitrate:
+            track.bitrate = parseText(data, fieldSize);
+            break;
+        case FieldId::SampleRate:
+            track.samplerate = parseText(data, fieldSize);
+            break;
+        case FieldId::Bpm: {
+            bool ok;
+            double bpm = parseText(data, fieldSize).toDouble(&ok);
+            if (ok) {
+                track.bpm = bpm;
+            }
+            break;
+        }
+        case FieldId::Comment:
+            track.comment = parseText(data, fieldSize);
+            break;
+        case FieldId::Grouping:
+            track.grouping = parseText(data, fieldSize);
+            break;
+        case FieldId::Label:
+            track.label = parseText(data, fieldSize);
+            break;
+        case FieldId::Year: {
+            bool ok;
+            int year = parseText(data, fieldSize).toInt(&ok);
+            if (ok) {
+                track.year = year;
+            }
+            break;
+        }
+        case FieldId::Key:
+            track.key = parseText(data, fieldSize);
```

Comment:
Please add a comment about the used key notation

Reply:
I'm not sure. I suppose it either whatever is stored in the ID3 tag, or depends on the notation configured in Serato.